src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -216,13 +212,8 @@ internal static bool TryGetBytesFromStream(Stream data, out ArraySegment<byte> b
             // than the buffer size. The default buffer size is 4KB, so this will incur a 4KB
             // allocation for any files less than 4KB. That's why, for example, the command
             // line compiler actually specifies a very small buffer size.
-            var success = stream.TryReadAll(buffer, 0, length) == length;
-
-            bytes = success
-                ? new ArraySegment<byte>(buffer)
-                : new ArraySegment<byte>(Array.Empty<byte>());
-
-            return success;
+            stream.ReadExactly(buffer, 0, length);
```

Comment:
This change technically changes behavior in that before if we failed to read all the file then we returned `false` but now we `throw`. That is true. However, also notice that `TryReadAll` does not reset the `Stream` position on failure and neither does this method. So before when `TryReadAll` only _partially_ read a file this method would exit with `Stream` in a corrupted state.

I did consider changing this method such that on `TryReadAll` failing invariants around file position were maintained. At the same time given we have no bug reports on the corrupted files, and `TryReadAll` should only fail if the underlying IO fails at which point we're gonna fail anyways, decided to just make this a more simple code path.